apps/fastload/boot.js

```diff
@@ -50,17 +51,55 @@ let fastload = function(n){
 };
 global.load = fastload;
 
+let appHistory = s.readJSON("fastload.history.json", true) || [];
+
+let resetHistory = function() {
+  appHistory = [];
+  s.writeJSON("fastload.history.json", appHistory);
+};
+
+let recordHistory = function() {
+  print("record: ", appHistory);
+  s.writeJSON("fastload.history.json", appHistory);
+};
+
+let initBootcdeRun;
+let setInitBootcdeRunCheck = function() {
+  initBootcdeRun = true;
+  print("init: ", initBootcdeRun);
+  load(".bootcde"); // fastload will not act on the initial load of ".bootcde" it seems (e.g. when exiting settings), at least not in the same way as subsequent calls. To work around this ".bootcde" is loaded once more on init. This makes the logic involving the bool `initBootdeRun` work as intended. // However, this seems to be somewhat unstable, causing errors when trying to write changes to "fastload.5.boot.js" ("Uncaught ReferenceError: "e" is not defined", "Uncaught Error: File already written with different data", "Uncaught Error: Too much data for file size"). // Why does exiting the settings generate the Espruino logo in the Web IDE console field while a long press of HW button does not? This difference seems to be a key to making this logic work without this extra load of ".bootcde" on init.
```

Comment:
@halemmerich do you have an idea regarding the comment on line 70 (after `load(".bootcde");` ?

Reply:
I've reverted this.